libtiledbsoma/src/utils/arrow_adapter.cc

```diff
@@ -978,85 +976,106 @@ ArraySchema ArrowAdapter::tiledb_schema_from_arrow_schema(
         bool isattr = true;
 
         for (int64_t i = 0; i < index_column_schema->n_children; ++i) {
-            auto achild = index_column_array->children[i];
-            auto schild = index_column_schema->children[i];
-            auto col_name = schild->name;
-            if (strcmp(child->name, col_name) == 0) {
-                if (ArrowAdapter::arrow_is_var_length_type(child->format)) {
-                    type = TILEDB_STRING_ASCII;
-                }
+            if (strcmp(child->name, index_column_schema->children[i]->name) ==
+                0) {
+                if (strcmp(child->name, "soma_geometry") == 0 &&
+                    spatial_column_info.first.get() != nullptr) {
+                    if (type_metadata.compare("WKB") != 0) {
+                        throw std::runtime_error(
+                            std::string(
+                                "Unkwown type metadata for `soma_geometry`. "
+                                "Expected 'WKB', found ") +
+                            std::string(type_metadata.data()));
+                    }
 
-                FilterList filter_list = ArrowAdapter::_create_dim_filter_list(
-                    child->name, platform_config, soma_type, ctx);
+                    for (int64_t j = 0;
+                         j < spatial_column_info.second->n_children;
+                         ++j) {
+                        auto min_dim = tiledb_dimension_from_arrow_schema(
+                            ctx,
+                            spatial_column_info.second->children[j],
+                            spatial_column_info.first->children[j],
+                            soma_type,
+                            type_metadata,
+                            "tiledb__internal__",
+                            "__min",
+                            platform_config);
+
+                        auto max_dim = tiledb_dimension_from_arrow_schema(
+                            ctx,
+                            spatial_column_info.second->children[j],
+                            spatial_column_info.first->children[j],
+                            soma_type,
+                            type_metadata,
+                            "tiledb__internal__",
+                            "__max",
+                            platform_config);
+
+                        dims.insert({min_dim.first.name(), min_dim.first});
+                        dims.insert({max_dim.first.name(), max_dim.first});
+
+                        use_current_domain &= min_dim.second;
+                        use_current_domain &= max_dim.second;
```

Comment:
For readability, I think this can be pulled out into a separate `_set_spatial_dimension` (or better name) helper function.